src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -30,10 +29,12 @@ public CSharpGenerateVariableService()
         }
 
         protected override bool IsExplicitInterfaceGeneration(SyntaxNode node)
-            => node is PropertyDeclarationSyntax;
+            => node.IsKind(SyntaxKind.PropertyDeclaration);
 
         protected override bool IsIdentifierNameGeneration(SyntaxNode node)
-            => node is IdentifierNameSyntax identifier && !identifier.Identifier.CouldBeKeyword();
+            => node.IsParentKind(SyntaxKind.ReturnStatement) ?
```

Comment:
Please place ? And : on their own line at the start of the line

Reply:
I'm a little confused why the return statement needs to be special cased. What is it about that construct that is causing it to be filtered out?

Reply:
> I'm a little confused why the return statement needs to be special cased

Basically, if we have just a random contextual keyword treated as an identifier, than it is probably not the thing, that user wants to see. However, if it is after `return`, than it is clear, that the user wants this name to be in context

Reply:
i think a better fix would be to adjust wherever we're doing the filtering and change it to whitelist certain contextual keywords that are unlikely to be syntactic constructs.  For example, `return from`, we would still not show it here, as the user is likely about to start a query expression.  But things like `field/property/etc.` aren't really syntactic constructs, they're just special semantic names that are only a problem in certain things (like attributes) so it's unlikely to be an issue.

This would be done regardless of if we're in a return-statement or not.

Reply:
Done. The list of "syntactic constructs" keywords is still a subject to change